cmd/asset-syncer/utils.go

```diff
@@ -222,10 +221,51 @@ type OCIRegistry struct {
 	*models.RepoInternal
 }
 
+func doReq(url, authHeader string) ([]byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("User-Agent", userAgent())
+	if len(authHeader) > 0 {
+		req.Header.Set("Authorization", authHeader)
+	}
+
+	res, err := netClient.Do(req)
+	if res != nil {
+		defer res.Body.Close()
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("request failed: %v", err)
+	}
+
+	return ioutil.ReadAll(res.Body)
+}
+
 // Checksum returns the sha256 of the repo
 func (r *OCIRegistry) Checksum() (string, error) {
-	// TBD
-	return "", nil
+	content := []byte{}
+	for _, appName := range r.repositories {
+		url, err := parseRepoURL(r.RepoInternal.URL)
+		if err != nil {
+			return "", err
+		}
+		// Retrieve the list of tags to add it to the list
+		// Caveat: Mutated image tags won't be detected as new
```

Comment:
Not necessary now, but I guess we could have a list of exceptions (`latest`) for which we actually grab the digest?

Reply:
In theory, `latest` should not be used for charts: https://github.com/helm/community/blob/master/hips/hip-0006.md#will-i-be-able-to-use-custom-tags-such-as-latest